crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -611,6 +618,34 @@ pub fn bridge_config(
         false => BridgeLocation::Mosquitto,
     };
     let mqtt_schema = MqttSchema::with_root(config.mqtt.topic_root.clone());
+    let proxy = config
+        .proxy
+        .address
+        .or_none()
+        .map(|address| {
+            let rustls_config = config.cloud_client_tls_config();
+            Ok::<_, ConfigError>(rumqttc::Proxy {
+                ty: match address.scheme() {
+                    ProxyScheme::Http => rumqttc::ProxyType::Http,
+                    ProxyScheme::Https => rumqttc::ProxyType::Https(
+                        rumqttc::TlsConfiguration::Rustls(Arc::new(rustls_config)),
+                    ),
+                },
+                addr: address.host().to_string(),
+                port: address.port().into(),
+                auth: match all_or_nothing((
+                    config.proxy.username.clone(),
+                    config.proxy.password.clone(),
+                ))
+                .map_err(|e| anyhow::anyhow!(e))?
+                {
+                    Some((username, password)) => rumqttc::ProxyAuth::Basic { username, password },
+                    None => rumqttc::ProxyAuth::None,
+                },
```

Comment:
`cargo fmt` is sometime confusing. I failed at first to read this as a regular `match` expression!